sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/_tests_/e2e/application-offering-change-request.institutions.controller.getEligibleApplications.e2e-spec.ts

```diff
@@ -0,0 +1,203 @@
+      applicationWithApprovedApplicationOfferingChange.applicationNumber =
+        "1000000001";
+      await db.application.save([
+        completedApplication,
+        archivedCompletedApplication,
+        applicationWithApprovedApplicationOfferingChange,
+      ]);
+
+      const endpoint = `/institutions/location/${collegeFLocation.id}/application-offering-change-request/available?page=0&pageLimit=10&sortField=applicationNumber&sortOrder=${FieldSortOrder.DESC}`;
+      const institutionUserToken = await getInstitutionToken(
+        InstitutionTokenTypes.CollegeFUser,
+      );
+
+      // Act/Assert
+      await request(app.getHttpServer())
+        .get(endpoint)
+        .auth(institutionUserToken, BEARER_AUTH_TYPE)
+        .expect(HttpStatus.OK)
+        .expect({
+          results: [
+            {
+              applicationNumber:
+                applicationWithApprovedApplicationOfferingChange.applicationNumber,
+              applicationId:
+                applicationWithApprovedApplicationOfferingChange.id,
+              studyStartDate:
+                applicationWithApprovedApplicationOfferingChange
+                  .currentAssessment.offering.studyStartDate,
+              studyEndDate:
+                applicationWithApprovedApplicationOfferingChange
+                  .currentAssessment.offering.studyEndDate,
+              fullName: getUserFullName(
+                applicationWithApprovedApplicationOfferingChange.student.user,
+              ),
+            },
+            {
+              applicationNumber: completedApplication.applicationNumber,
+              applicationId: completedApplication.id,
+              studyStartDate:
+                completedApplication.currentAssessment.offering.studyStartDate,
+              studyEndDate:
+                completedApplication.currentAssessment.offering.studyEndDate,
+              fullName: getUserFullName(completedApplication.student.user),
+            },
+          ],
+          count: 2,
+        });
+    },
+  );
+
+  it("Should return only matching available eligible applications that can be requested for an application offering change for the location when searched.", async () => {
+    // Arrange
+
+    // Student 1 has a completed application to the institution.
+    const completedApplication = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { institutionLocation: collegeFLocation },
+      {
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+    // Student 2 has a completed application to the institution.
+    await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { institutionLocation: collegeFLocation },
+      {
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+
+    const endpoint = `/institutions/location/${collegeFLocation.id}/application-offering-change-request/available?page=0&pageLimit=10&searchCriteria=${completedApplication.applicationNumber}`;
```

Comment:
Just checking. Do we feel it is ok to skip the test case for search by student name because it is done in other e2e test(getCompletedApplications) and they share the same service?

Reply:
actually, `available`. use one service and `in progress and `completed` share one service. so, IMO. I think it's needed.